### DISCUSSION TITLE
S3 sync support?

### DISCUSSION THREAD

**Discussion by tgreene** [joplin-vieweb/joplin-vieweb#65] [Ideas]
Any thoughts about adding support for other sync sources, esp. S3? 

Thanks!

**Comment by gri38** [comment]
Hello.
I've just checked and it's supported by Joplin terminal application, so yes I can add it.
I'll do one of this days.

Followthis issue: https://github.com/joplin-vieweb/joplin-vieweb/issues/61 to be notified when it's resolved.

**Comment by gri38** [comment]
BTW, @tgreene 
The doc of joplin terminal lists those settings for configuring S3:

    sync.8.path                    AWS S3 bucket.
                                    Attention: If you change this location,
                                    make sure you copy all your content to it
                                    before syncing, otherwise all files will be
                                    removed! See the FAQ for more details:
                                    https://joplinapp.org/faq/
                                    Type: string.

    sync.8.url                     AWS S3 URL.
                                    Type: string.
                                    Default: "https://s3.amazonaws.com/"

    sync.8.region                  AWS region.
                                    Type: string.

    sync.8.username                AWS access key.
                                    Type: string.

    sync.8.password                AWS secret key.
                                    Type: string.

    sync.8.forcePathStyle          Force path style.
                                    Type: bool.
                                    Default: false


I don't know S3, I don't have an account.

According to you and your account: do you need all the parameters or not?
If for instance only url, username and password are required, I can do that very quickly.

Let me know.

**Reply by tgreene** [reply]
Hi @gri38 thanks for the fast response!  AFAIK, region is required, however I think it can be infered from the URL in most cases.

For example, if the URL is "https://s3.us-west-2.amazonaws.com" then the region is "us-west-2"

That said, I think for some older S3 buckets or custom configs on S3, it may not always be a guaranteed way to infer the region.

**Reply by gri38** [reply]
Hello @tgreene 
I've released https://github.com/joplin-vieweb/joplin-vieweb/releases/tag/V2.5.0 
Could you please try: I tested only the good configuration of joplin terminal, but not the true sync with a S3 bucket.

**Comment by gri38** [comment]
implemented in release [2.5.1](https://github.com/joplin-vieweb/joplin-vieweb/releases/tag/v2.5.1). Please open again if it doesn't work as expected.
